### PR TITLE
Generate a nice authors.md file

### PR DESCRIPTION
**Pull request by certik**
This is important and once the article gets accepted, we should put this table into the README, so that anyone can easily check that all authors satisfy the authorship criteria, as well as to see what each author actually did in the article.

I didn't commit the generated file for now, so that we don't have conflicts. Once all authors are in, I suggest we commit the generated file, so that it's easy to see on GitHub. See my comment below how it looks like.

### CODE REVIEW

**Comment by certik** [comment]
# Article Authors

All authors satisfy the following ICMJE [criteria](http://www.icmje.org/recommendations/browse/roles-and-responsibilities/defining-the-role-of-authors-and-contributors.html#two) for authorship:
1. Substantial contributions to the conception or design of the work; or the
   acquisition, analysis, or interpretation of data for the work; AND
2. Drafting the work or revising it critically for important intellectual
   content; AND
3. Final approval of the version to be published; AND
4. Agreement to be accountable for all aspects of the work in ensuring that
   questions related to the accuracy or integrity of any part of the work are
   appropriately investigated and resolved.

The column `SymPy Commits` lists how many commits the authors contributed to
SymPy (with a link to the actual commits), which shows how the requirement 1.
is satisfied.

The column `Article Contributions` lists all commits and comments to the
article itself, which shows how the requirement 2. is satisfied.

All authors approved the final version of the article, satisfying the
requirement 3.

Finally, all authors agreed to the requirement 4.

| Author | GitHub Profile | SymPy Commits | Article Contributions | Affiliation |
| :-- | :-- | :-: | :-- | :-- |
| Aaron Meurer | [@asmeurer](https://github.com/asmeurer) | [2636](https://github.com/sympy/sympy/commits?author=asmeurer) | [commits](https://github.com/sympy/sympy-paper/commits?author=asmeurer), [comments](https://github.com/sympy/sympy-paper/search?q=asmeurer&type=Issues) | University of South Carolina, Columbia, SC 29201 |
| Ondřej Čertík | [@certik](https://github.com/certik) | [1443](https://github.com/sympy/sympy/commits?author=certik) | [commits](https://github.com/sympy/sympy-paper/commits?author=certik), [comments](https://github.com/sympy/sympy-paper/search?q=certik&type=Issues) | Los Alamos National Laboratory, Los Alamos, NM 87545 |
| AMiT Kumar | [@aktech](https://github.com/aktech) | [263](https://github.com/sympy/sympy/commits?author=aktech) | [commits](https://github.com/sympy/sympy-paper/commits?author=aktech), [comments](https://github.com/sympy/sympy-paper/search?q=aktech&type=Issues) | Delhi Technological University, Shahbad Daulatpur, Bawana Road, New Delhi 110042, India |
| Jason K. Moore | [@moorepants](https://github.com/moorepants) | [253](https://github.com/sympy/sympy/commits?author=moorepants) | [commits](https://github.com/sympy/sympy-paper/commits?author=moorepants), [comments](https://github.com/sympy/sympy-paper/search?q=moorepants&type=Issues) | University of California, Davis, Davis, CA 95616 |
| Sartaj Singh | [@leosartaj](https://github.com/leosartaj) | [199](https://github.com/sympy/sympy/commits?author=leosartaj) | [commits](https://github.com/sympy/sympy-paper/commits?author=leosartaj), [comments](https://github.com/sympy/sympy-paper/search?q=leosartaj&type=Issues) | Indian Institute of Technology (BHU), Varanasi, Uttar Pradesh 221005, India |
| Thilina Rathnayake | [@thilinarmtb](https://github.com/thilinarmtb) | [186](https://github.com/sympy/sympy/commits?author=thilinarmtb) | [commits](https://github.com/sympy/sympy-paper/commits?author=thilinarmtb), [comments](https://github.com/sympy/sympy-paper/search?q=thilinarmtb&type=Issues) | University of Moratuwa, Bandaranayake Mawatha, Katubedda, Moratuwa 10400, Sri Lanka |
| Brian E. Granger | [@ellisonbg](https://github.com/ellisonbg) | [160](https://github.com/sympy/sympy/commits?author=ellisonbg) | [commits](https://github.com/sympy/sympy-paper/commits?author=ellisonbg), [comments](https://github.com/sympy/sympy-paper/search?q=ellisonbg&type=Issues) | California Polytechnic State University, San Luis Obispo, CA 93407 |
| Richard P. Muller | [@rpmuller](https://github.com/rpmuller) | [158](https://github.com/sympy/sympy/commits?author=rpmuller) | [commits](https://github.com/sympy/sympy-paper/commits?author=rpmuller), [comments](https://github.com/sympy/sympy-paper/search?q=rpmuller&type=Issues) | Center for Computing Research, Sandia National Laboratories, Albuquerque, NM 87185 |
| Francesco Bonazzi | [@Upabjojr](https://github.com/Upabjojr) | [150](https://github.com/sympy/sympy/commits?author=Upabjojr) | [commits](https://github.com/sympy/sympy-paper/commits?author=Upabjojr), [comments](https://github.com/sympy/sympy-paper/search?q=Upabjojr&type=Issues) | Max Planck Institute of Colloids and Interfaces, Am Mühlenberg 1, 14476 Potsdam, Germany |
| Harsh Gupta | [@hargup](https://github.com/hargup) | [123](https://github.com/sympy/sympy/commits?author=hargup) | [commits](https://github.com/sympy/sympy-paper/commits?author=hargup), [comments](https://github.com/sympy/sympy-paper/search?q=hargup&type=Issues) | Indian Institute of Technology Kharagpur, Kharagpur, West Bengal 721302, India |
| Shivam Vats | [@shivamvats](https://github.com/shivamvats) | [122](https://github.com/sympy/sympy/commits?author=shivamvats) | [commits](https://github.com/sympy/sympy-paper/commits?author=shivamvats), [comments](https://github.com/sympy/sympy-paper/search?q=shivamvats&type=Issues) | Indian Institute of Technology Kharagpur, Kharagpur, West Bengal 721302, India |
| Fredrik Johansson | [@fredrik-johansson](https://github.com/fredrik-johansson) | [119](https://github.com/sympy/sympy/commits?author=fredrik-johansson) | [commits](https://github.com/sympy/sympy-paper/commits?author=fredrik-johansson), [comments](https://github.com/sympy/sympy-paper/search?q=fredrik-johansson&type=Issues) | INRIA Bordeaux-Sud-Ouest - LFANT project-team, 200 Avenue de la Vieille Tour, 33405 Talence, France |
| Fabian Pedregosa | [@fabianp](https://github.com/fabianp) | [114](https://github.com/sympy/sympy/commits?author=fabianp) | [commits](https://github.com/sympy/sympy-paper/commits?author=fabianp), [comments](https://github.com/sympy/sympy-paper/search?q=fabianp&type=Issues) | INRIA - SIERRA project-team, 2 Rue Simone IFF, 75012 Paris, France |
| Ashutosh Saboo | [@ashutoshsaboo](https://github.com/ashutoshsaboo) | [17](https://github.com/sympy/sympy/commits?author=ashutoshsaboo) | [commits](https://github.com/sympy/sympy-paper/commits?author=ashutoshsaboo), [comments](https://github.com/sympy/sympy-paper/search?q=ashutoshsaboo&type=Issues) | Birla Institute of Technology and Science, Pilani, K.K. Birla Goa Campus, NH 17B Bypass Road, Zuarinagar, Sancoale, Goa 403726, India |
| Isuru Fernando | [@isuruf](https://github.com/isuruf) | [7](https://github.com/sympy/sympy/commits?author=isuruf) | [commits](https://github.com/sympy/sympy-paper/commits?author=isuruf), [comments](https://github.com/sympy/sympy-paper/search?q=isuruf&type=Issues) | University of Moratuwa, Bandaranayake Mawatha, Katubedda, Moratuwa 10400, Sri Lanka |
| Sumith | [@Sumith1896](https://github.com/Sumith1896) | [6](https://github.com/sympy/sympy/commits?author=Sumith1896) | [commits](https://github.com/sympy/sympy-paper/commits?author=Sumith1896), [comments](https://github.com/sympy/sympy-paper/search?q=Sumith1896&type=Issues) | Indian Institute of Technology Bombay, Powai, Mumbai, Maharashtra 400076, India |
| Robert Cimrman | [@rc](https://github.com/rc) | [1](https://github.com/sympy/sympy/commits?author=rc) | [commits](https://github.com/sympy/sympy-paper/commits?author=rc), [comments](https://github.com/sympy/sympy-paper/search?q=rc&type=Issues) | New Technologies - Research Centre, University of West Bohemia, Univerzitní 8, 306 14 Plzeň, Czech Republic |

Note: This file was automatically generated from `authors.json` using `list_text.py`

File: authors/list_text.py [hunk 1]
```diff
@@ -2,18 +2,66 @@
...
+            "Article Contributions | Affiliation |\n")
+    f.write(u"| :---   | :---           |     :---:     | "
+            ":---                  | :---        |\n")
+
+    for author in author_list:
+        f.write(u"| %s | [@%s](https://github.com/%s) | "
+            "[%d](https://github.com/sympy/sympy/commits?author=%s) | "
+            "[commits](https://github.com/sympy/sympy-paper/commits?author=%s),"
+                " [comments](https://github.com/sympy/sympy-paper/search"
+                "?q=%s&type=Issues) "
```
Comment (asmeurer) [inline authors/list_text.py]: For comments you should search for "commenter:asmeurer" (see https://help.github.com/articles/searching-issues/).

Comment (certik) [inline authors/list_text.py]: Yes, I was going to ask how to fix this link. I am also fixing another issue, will push commits soon.

**Comment by asmeurer** [comment]
Thanks. This will also help when I submit, since I have to copy and paste all author information into a form.

**Comment by certik** [comment]
@asmeurer I was planning adding another script just for the form, but if you think this table is enough, then we can use this table of course.

**Comment by asmeurer** [comment]
If you want to script the form that would be awesome too :)

File: authors/list_text.py (hunk 1 above)
Comment (certik) [inline authors/list_text.py]: Fixed in ec0e97ce5e85ee4c5b9e9aa066d1ece391f43dc3, thanks @asmeurer for this.

**Comment by certik** [comment]
As of e367837f2fb65650b892f4695612e1e0a24c2855:

# Article Authors

All authors satisfy the following ICMJE [criteria](http://www.icmje.org/recommendations/browse/roles-and-responsibilities/defining-the-role-of-authors-and-contributors.html#two) for authorship:
1. Substantial contributions to the conception or design of the work; or the
   acquisition, analysis, or interpretation of data for the work; AND
2. Drafting the work or revising it critically for important intellectual
   content; AND
3. Final approval of the version to be published; AND
4. Agreement to be accountable for all aspects of the work in ensuring that
   questions related to the accuracy or integrity of any part of the work are
   appropriately investigated and resolved.

The column `SymPy Commits` lists how many commits the authors contributed to
SymPy (with a link to the actual commits), which shows how the requirement 1.
is satisfied.

The column `Article Contributions` lists all commits and comments to the
article itself, which shows how the requirement 2. is satisfied.

All authors approved the final version of the article, satisfying the
requirement 3.

Finally, all authors agreed to the requirement 4.

| Author | GitHub Profile | SymPy Commits | Article Contributions | Affiliation |
| :-- | :-- | :-: | :-- | :-- |
| Aaron Meurer | [@asmeurer](https://github.com/asmeurer) | [2636](https://github.com/sympy/sympy/commits?author=asmeurer) | [commits](https://github.com/sympy/sympy-paper/commits?author=asmeurer), [comments](https://github.com/sympy/sympy-paper/search?q=commenter:asmeurer&type=Issues) | University of South Carolina, Columbia, SC 29201 |
| Ondřej Čertík | [@certik](https://github.com/certik) | [1443](https://github.com/sympy/sympy/commits?author=certik) | [commits](https://github.com/sympy/sympy-paper/commits?author=certik), [comments](https://github.com/sympy/sympy-paper/search?q=commenter:certik&type=Issues) | Los Alamos National Laboratory, Los Alamos, NM 87545 |
| AMiT Kumar | [@aktech](https://github.com/aktech) | [263](https://github.com/sympy/sympy/commits?author=aktech) | [commits](https://github.com/sympy/sympy-paper/commits?author=aktech), [comments](https://github.com/sympy/sympy-paper/search?q=commenter:aktech&type=Issues) | Delhi Technological University, Shahbad Daulatpur, Bawana Road, New Delhi 110042, India |
| Jason K. Moore | [@moorepants](https://github.com/moorepants) | [253](https://github.com/sympy/sympy/commits?author=moorepants) | [commits](https://github.com/sympy/sympy-paper/commits?author=moorepants), [comments](https://github.com/sympy/sympy-paper/search?q=commenter:moorepants&type=Issues) | University of California, Davis, Davis, CA 95616 |
| Sartaj Singh | [@leosartaj](https://github.com/leosartaj) | [199](https://github.com/sympy/sympy/commits?author=leosartaj) | [commits](https://github.com/sympy/sympy-paper/commits?author=leosartaj), [comments](https://github.com/sympy/sympy-paper/search?q=commenter:leosartaj&type=Issues) | Indian Institute of Technology (BHU), Varanasi, Uttar Pradesh 221005, India |
| Thilina Rathnayake | [@thilinarmtb](https://github.com/thilinarmtb) | [186](https://github.com/sympy/sympy/commits?author=thilinarmtb) | [commits](https://github.com/sympy/sympy-paper/commits?author=thilinarmtb), [comments](https://github.com/sympy/sympy-paper/search?q=commenter:thilinarmtb&type=Issues) | University of Moratuwa, Bandaranayake Mawatha, Katubedda, Moratuwa 10400, Sri Lanka |
| Brian E. Granger | [@ellisonbg](https://github.com/ellisonbg) | [160](https://github.com/sympy/sympy/commits?author=ellisonbg) | [commits](https://github.com/sympy/sympy-paper/commits?author=ellisonbg), [comments](https://github.com/sympy/sympy-paper/search?q=commenter:ellisonbg&type=Issues) | California Polytechnic State University, San Luis Obispo, CA 93407 |
| Richard P. Muller | [@rpmuller](https://github.com/rpmuller) | [158](https://github.com/sympy/sympy/commits?author=rpmuller) | [commits](https://github.com/sympy/sympy-paper/commits?author=rpmuller), [comments](https://github.com/sympy/sympy-paper/search?q=commenter:rpmuller&type=Issues) | Center for Computing Research, Sandia National Laboratories, Albuquerque, NM 87185 |
| Francesco Bonazzi | [@Upabjojr](https://github.com/Upabjojr) | [150](https://github.com/sympy/sympy/commits?author=Upabjojr) | [commits](https://github.com/sympy/sympy-paper/commits?author=Upabjojr), [comments](https://github.com/sympy/sympy-paper/search?q=commenter:Upabjojr&type=Issues) | Max Planck Institute of Colloids and Interfaces, Am Mühlenberg 1, 14476 Potsdam, Germany |
| Harsh Gupta | [@hargup](https://github.com/hargup) | [123](https://github.com/sympy/sympy/commits?author=hargup) | [commits](https://github.com/sympy/sympy-paper/commits?author=hargup), [comments](https://github.com/sympy/sympy-paper/search?q=commenter:hargup&type=Issues) | Indian Institute of Technology Kharagpur, Kharagpur, West Bengal 721302, India |
| Shivam Vats | [@shivamvats](https://github.com/shivamvats) | [122](https://github.com/sympy/sympy/commits?author=shivamvats) | [commits](https://github.com/sympy/sympy-paper/commits?author=shivamvats), [comments](https://github.com/sympy/sympy-paper/search?q=commenter:shivamvats&type=Issues) | Indian Institute of Technology Kharagpur, Kharagpur, West Bengal 721302, India |
| Fredrik Johansson | [@fredrik-johansson](https://github.com/fredrik-johansson) | [119](https://github.com/sympy/sympy/commits?author=fredrik-johansson) | [commits](https://github.com/sympy/sympy-paper/commits?author=fredrik-johansson), [comments](https://github.com/sympy/sympy-paper/search?q=commenter:fredrik-johansson&type=Issues) | INRIA Bordeaux-Sud-Ouest - LFANT project-team, 200 Avenue de la Vieille Tour, 33405 Talence, France |
| Fabian Pedregosa | [@fabianp](https://github.com/fabianp) | [114](https://github.com/sympy/sympy/commits?author=fabianp) | [commits](https://github.com/sympy/sympy-paper/commits?author=fabianp), [comments](https://github.com/sympy/sympy-paper/search?q=commenter:fabianp&type=Issues) | INRIA - SIERRA project-team, 2 Rue Simone IFF, 75012 Paris, France |
| Ashutosh Saboo | [@ashutoshsaboo](https://github.com/ashutoshsaboo) | [17](https://github.com/sympy/sympy/commits?author=ashutoshsaboo) | [commits](https://github.com/sympy/sympy-paper/commits?author=ashutoshsaboo), [comments](https://github.com/sympy/sympy-paper/search?q=commenter:ashutoshsaboo&type=Issues) | Birla Institute of Technology and Science, Pilani, K.K. Birla Goa Campus, NH 17B Bypass Road, Zuarinagar, Sancoale, Goa 403726, India |
| Isuru Fernando | [@isuruf](https://github.com/isuruf) | [7](https://github.com/sympy/sympy/commits?author=isuruf) | [commits](https://github.com/sympy/sympy-paper/commits?author=isuruf), [comments](https://github.com/sympy/sympy-paper/search?q=commenter:isuruf&type=Issues) | University of Moratuwa, Bandaranayake Mawatha, Katubedda, Moratuwa 10400, Sri Lanka |
| Sumith | [@Sumith1896](https://github.com/Sumith1896) | [6](https://github.com/sympy/sympy/commits?author=Sumith1896) | [commits](https://github.com/sympy/sympy-paper/commits?author=Sumith1896), [comments](https://github.com/sympy/sympy-paper/search?q=commenter:Sumith1896&type=Issues) | Indian Institute of Technology Bombay, Powai, Mumbai, Maharashtra 400076, India |
| Robert Cimrman | [@rc](https://github.com/rc) | [1](https://github.com/sympy/sympy/commits?author=robert.cimrman@gmail.com) | [commits](https://github.com/sympy/sympy-paper/commits?author=rc), [comments](https://github.com/sympy/sympy-paper/search?q=commenter:rc&type=Issues) | New Technologies - Research Centre, University of West Bohemia, Univerzitní 8, 306 14 Plzeň, Czech Republic |

Note: This file was automatically generated from `authors.json` using `list_text.py`.

**Comment by certik** [comment]
@asmeurer how does the form look like --- how can it be scripted?

I have to include all authors into our internal form as well, and I am planning to simply write a script that prints out all the required info in the correct form and order of the fields in the form, so that I can just easily copy & paste it, as well as check that things are correct.

**Comment by asmeurer** [comment]
screen shot 2016-05-18 at 5 30 11 pm

See http://sisc.siam.org/.

**Comment by certik** [comment]
@asmeurer thanks. I have no idea how to script this, besides hacking the JavaScript, which I think is now worth the time.

To save time, we might want to add the following fields into `authors.json`: `siam_address_line`, `siam_city`, `siam_country` --- anyone can add these in as a PR, since all the information is already there, just not in the right form. Then we can write a script that prints the information exactly in the form and order as is needed, and you just copy & paste.

The First, Middle and Last name can probably be inferred automatically from `name`.

I would skip Person Title.

**Pull request merged**